pkg/operator/operator/k8s_specs.go

```diff
@@ -532,62 +561,210 @@ func getEnvVars(api *spec.API) []kcore.EnvVar {
+		envVars = append(envVars,
+			kcore.EnvVar{
+				Name: "HOST_IP",
+				ValueFrom: &kcore.EnvVarSource{
+					FieldRef: &kcore.ObjectFieldSelector{
+						FieldPath: "status.hostIP",
+					},
 				},
 			},
+			kcore.EnvVar{
+				Name:  "CORTEX_WORKERS_PER_REPLICA",
+				Value: s.Int32(api.Autoscaling.WorkersPerReplica),
+			},
+			kcore.EnvVar{
+				Name:  "CORTEX_THREADS_PER_WORKER",
+				Value: s.Int32(api.Autoscaling.ThreadsPerWorker),
+			},
+			kcore.EnvVar{
+				Name:  "CORTEX_MAX_REPLICA_CONCURRENCY",
+				Value: s.Int64(api.Autoscaling.MaxReplicaConcurrency),
+			},
+			kcore.EnvVar{
+				Name: "CORTEX_MAX_WORKER_CONCURRENCY",
+				// add 1 because it was required to achieve the target concurrency for 1 worker, 1 thread
+				Value: s.Int64(1 + int64(math.Round(float64(api.Autoscaling.MaxReplicaConcurrency)/float64(api.Autoscaling.WorkersPerReplica)))),
+			},
+			kcore.EnvVar{
+				Name:  "CORTEX_SO_MAX_CONN",
+				Value: s.Int64(api.Autoscaling.MaxReplicaConcurrency + 100), // add a buffer to be safe
+			},
+			kcore.EnvVar{
+				Name:  "CORTEX_SERVING_PORT",
+				Value: _defaultPortStr,
+			},
+			kcore.EnvVar{
+				Name:  "CORTEX_API_SPEC",
+				Value: aws.S3Path(config.Cluster.Bucket, api.Key),
+			},
+			kcore.EnvVar{
+				Name:  "CORTEX_CACHE_DIR",
+				Value: _specCacheDir,
+			},
+			kcore.EnvVar{
+				Name:  "CORTEX_PROJECT_DIR",
+				Value: path.Join(_emptyDirMountPath, "project"),
+			},
+		)
+
+		if api.Predictor.PythonPath != nil {
+			envVars = append(envVars, kcore.EnvVar{
+				Name:  "PYTHON_PATH",
+				Value: path.Join(_emptyDirMountPath, "project", *api.Predictor.PythonPath),
+			})
+		}
+
+		if api.Predictor.Type == userconfig.ONNXPredictorType {
+			envVars = append(envVars, kcore.EnvVar{
+				Name:  "CORTEX_MODEL_DIR",
+				Value: path.Join(_emptyDirMountPath, "model"),
+			})
+		}
+
+		if api.Predictor.Type == userconfig.TensorFlowPredictorType {
+			envVars = append(envVars, []kcore.EnvVar{
+				{
+					Name:  "CORTEX_MODEL_DIR",
+					Value: path.Join(_emptyDirMountPath, "model"),
+				},
+				{
+					Name:  "CORTEX_TF_BASE_SERVING_PORT",
+					Value: _tfBaseServingPortStr,
+				},
+			}...)
+		}
+	}
+
+	if api.Compute.ASIC > 0 {
+		if (container == _apiContainerName && api.Predictor.Type != userconfig.TensorFlowPredictorType) ||
+			(container == _tfServingContainerName && api.Predictor.Type == userconfig.TensorFlowPredictorType) {
+			envVars = append(envVars,
+				kcore.EnvVar{
+					Name:  "NEURONCORE_GROUP_SIZES",
+					Value: s.Int64(api.Compute.ASIC * _coresPerASIC / int64(api.Autoscaling.WorkersPerReplica)),
+				},
+				kcore.EnvVar{
+					Name:  "NEURON_RTD_ADDRESS",
+					Value: fmt.Sprintf("unix:%s", _neuronRTDSocketReadinessFile),
+				},
+			)
+		}
+
+		if container == _tfServingContainerName && api.Predictor.Type == userconfig.TensorFlowPredictorType {
+			envVars = append(envVars, []kcore.EnvVar{
+				{
+					Name:  "TF_WORKERS",
+					Value: s.Int32(api.Autoscaling.WorkersPerReplica),
```

Comment:
Should this be set regardless of `api.Compute.ASIC`?

Reply:
Nope. Marking this as solved as it has already been discussed.